src/stream/src/common/table/state_table.rs

```diff
@@ -896,8 +892,8 @@ where
 
                     #[for_await]
                     for entry in merged_stream.take(self.watermark_cache.capacity()) {
-                        let (pk, _row) = entry?;
-                        let (_, pk) = deserialize_pk_with_vnode(&pk[..], &self.pk_serde)?;
+                        let keyed_row = entry?;
+                        let (_, pk) = deserialize_pk_with_vnode(keyed_row.key(), &self.pk_serde)?;
```

Comment:
I guess `deserialize_pk_with_vnode` call here is not needed any more.

Reply:
Good catch.